boards/shields/x_nucleo_iks02a1/boards/nucleo_f411re.defconfig

```diff
@@ -0,0 +1,30 @@
+# Copyright (c) 2020 STMicroelectronics
+#
+# SPDX-License-Identifier: Apache-2.0
+
+if BOARD_NUCLEO_F411RE
+
+if I2S
+
+config I2S_STM32
+	default y
+
+# configure PLLI2S to generate a I2SxCLK=128MHz
+config I2S_STM32_USE_PLLI2S_ENABLE
+	default y
+
+config I2S_STM32_PLLI2S_PLLM
+	default 8
+
+config I2S_STM32_PLLI2S_PLLN
+	default 192
+
+config I2S_STM32_PLLI2S_PLLR
+	default 3
+
+config I2S_1
```

Comment:
Actually I have some doubts here. I explain to you, I think that which I2S should be enabled does not depend by board (nucleo_f411re does not need any I2S to be enabled because it does not have any audio h/w capability) but by shield. The x_nucleo_f411re shield has a microphone exported onto I2S_1 of the hosting board.

Reply:
If I put it in board I should enable all 5 I2S, which is not good.
Actually It was done like this before, but now it is good that the shield part can overload the defconfig as well. So, I can define only I2S_1, which is the only one needed for x_nucleo_iks02a1

Reply:
> If I put it in board I should enable all 5 I2S, which is not good.

I don't see why. You can just enable I2S_1. This is what we are doing for other peripherals.

>  now it is good that the shield part can overload the defconfig as well

I get this point. Yet, this is in line with what you are doing when enabling setting i2s1 as arduino_i2s and enabling it.
This is just a default, conditional, configuration and this is the same we're doing with other peripherals like i2c and spi. I don't see a reason why it should be different for i2s.

Reply:
OK. What happens in case a different shield requires I2S_2 or I2S_5 for example?

Reply:
You'll not use arduino_i2s in this case.
Shields are essentially defined by the connector they use. You're using an arduino shield, connected to what is defined as arduino_i2s, which is I2S_1 in the nucleo_f411re case (can't be changed, this is defined by a fixed set of pins).
Another shield might use another connector, or might use different pins to access SPI, but then you'll have to define other connector on board side.

Reply:
OK, got your point! Actually I misunderstood completely your point until now.